templates/node.js-express-webapp-on-linux.yml

```diff
@@ -6,35 +6,74 @@
 trigger:
 - master
 
-pool:
-  vmImage: 'ubuntu-latest'
-
-steps:
-- task: NodeTool@0
-  inputs:
-    versionSpec: '10.x'
-  displayName: 'Install Node.js'
-
-- script: |
-    npm install
-    npm build
-  displayName: 'npm install and build'
-
-- task: ArchiveFiles@2
-  displayName: 'Archive files'
-  inputs:
-    rootFolderOrFile: '$(System.DefaultWorkingDirectory)'
-    includeRootFolder: false
-    archiveType: zip
-    archiveFile: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
-    replaceExistingArchive: true
+variables:
+
+  # Azure Resource Manager connection created during pipeline creation
+  azureSubscription: '{{ azureRmConnection.Name }}'
+  

```

Comment:
- Move this to deploy stage and 
- add `PublishPipelineArtifact@0` task or `upload` macro in the CI job (whichever is deployed)
- the web app task should be in `deployment` job
- Temporarily, we have to include `DownloadPipelineArtifact@1` task to download the artifact. We will remove this when @kasubram, @niadak PR for auto-injection of download steps, and checkout, download macro support is added to the `deployment` job
- for `environment`: we can use `development` for now, @chshrikh is checking feasibility of supporting variables, if we can pull it through then we can use a variable for the format $(build.repository.name)-dev or something similar
@kmkumaran, @azooinmyluggage, @ashokirla  fyi

Reply:
Above comments apply to all templates

Reply:
Updated the PR.